### ISSUE TITLE
Update to FastRoute 0.6.0

### ISSUE THREAD

**Issue by jbrooksuk** [laravel/lumen-framework#178]
https://github.com/nikic/FastRoute/releases/tag/v0.6.0

Optional routes is a big'n here.

**Comment by GrahamCampbell** [comment]
We'll probably have to do this on 5.2 rather than 5.1 since there are breaking changes inside fast route.

**Comment by GrahamCampbell** [comment]
For example, if somone had cached their routes that cache is broken on upgrade and needs manually clearing.

**Comment by jasonlewis** [comment]
You can't cache routes in Lumen though.

**Comment by GrahamCampbell** [comment]
> You can't cache routes in Lumen though.

True out of the box I guess. I'm not against this, I just think we need to be careful not to break things. :)

**Comment by jasonlewis** [comment]
Just a quick flick through and it doesn't seem like there would be any compatibility issues ASSUMING people aren't manually caching their routes. Although I don't even think it'd be possible because of how the routes are registered with the FastRoute dispatcher.

Need to test it a bit more thoroughly though.

**Comment by orditeck** [comment]
Is there a way to use 0.6.0 if I want to? I tried to add `"nikic/fast-route": "0.6.*"` in my composer.json but it returns an error:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laravel/lumen-framework v5.1.3 requires nikic/fast-route 0.4.* -> no matching package found.
    - laravel/lumen-framework v5.1.2 requires nikic/fast-route 0.4.* -> no matching package found.
    - laravel/lumen-framework v5.1.1 requires nikic/fast-route 0.4.* -> no matching package found.
    - laravel/lumen-framework v5.1.0 requires nikic/fast-route 0.4.* -> no matching package found.
    - laravel/lumen-framework 5.1.x-dev requires nikic/fast-route 0.4.* -> no matching package found.
    - laravel/lumen-framework v5.1.3 requires nikic/fast-route 0.4.* -> no matching package found.
    - Installation request for laravel/lumen-framework 5.1.* -> satisfiable by laravel/lumen-framework[5.1.x-dev, v5.1.0, v5.1.1, v5.1.2, v5.1.3].

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
```

**Comment by GrahamCampbell** [comment]
> Is there a way to use 0.6.0 if I want to? I tried to add "nikic/fast-route": "0.6.*" in my composer.json 

Yeh, you can't add your own versions like that.

**Comment by GrahamCampbell** [comment]
> Edit: just needed to use "nikic/fast-route": "0.6.0 as 0.4.0", can now use $app->get('rooms/{method:add|edit}[/{id}]' :smile:

I wouldn't recommend doing that.

**Comment by GrahamCampbell** [comment]
Lumen 5.2 will ship with fast route 0.6.

**Comment by jbrooksuk** [comment]
:boom:

**Comment by GrahamCampbell** [comment]
Or 0.7 if it's released before Lumen v5.2.0 gets tagged.
